modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/field/customizer/SegmentsContextVocabularySegmentsFieldCustomizer.java

```diff
@@ -93,29 +93,38 @@ public List<Field.Option> getOptions(Locale locale) {
 
 		Group group = _groupLocalService.fetchCompanyGroup(companyId);
 
-		return Optional.ofNullable(
+		AssetVocabulary groupVocabulary =
 			_assetVocabularyLocalService.fetchGroupVocabulary(
-				group.getGroupId(), assetVocabulary)
-		).map(
-			AssetVocabulary::getCategories
-		).orElseGet(
-			() -> {
-				if (_log.isWarnEnabled()) {
-					_log.warn(
-						StringBundler.concat(
-							"No vocabulary was found with name ",
-							assetVocabulary, " in company ", companyId));
-				}
-
-				return Collections.emptyList();
+				group.getGroupId(), assetVocabulary);
+
+		if (groupVocabulary == null) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					StringBundler.concat(
+						"No vocabulary was found with name ", assetVocabulary,
+						" in company ", companyId));
 			}
-		).stream(
-		).map(
+
+			return Collections.emptyList();
+		}
+
+		List<AssetCategory> assetCategories = groupVocabulary.getCategories();
+
+		if (assetCategories == null) {
```

Comment:
@JanisZhang I do not think you need this null checking as it can not be null, please make sure you verify it can not be null